Clarifying SNS asset canister documentation (#1965)
dfinity/portal: docs/developer-docs/integrations/sns/managing/sns-asset-canister.md

The general overview of deploying an asset canister during an SNS launch is as follows:
- First, the asset canister must be created with or upgraded to a Wasm file from [dfx 0.14.1+](https://github.com/dfinity/sdk/blob/release-0.14.1/src/distributed/assetstorage.wasm.gz).

> - Then, the dapp developers hand over control of the entire dapp to the NNS, including the asset canister.

Reviewer: The line above uses a hard-coded wasm that will become outdated pretty quickly, and it won't work with the last step below. I think the last paragraph of this page would be perfect for this bullet point

Author: Updated this line per your feedback.

- A single NNS proposal creates the dapp's SNS canisters, initializes them, and hands the dapp canisters over to the SNS. The decentralization swap begins.

> - The SNS community then upgrades the dapp canister through an SNS proposal and sets the privileges in the dapp. 

Reviewer: What dapp canister is this talking about?

Author: Added clarification

> - After the SNS is created, the asset canister can be upgraded via upgrade arguments that specify the SNS governance's canister ID. 

Reviewer: I think this needs a 'why'

Author: @LaraAS Could you help provide a 'why' for this bullet point?

Reviewer 2: My quick version of the why:
* There are certain functions that must only be triggered by a decisions of the SNS DAO.
* To realise this in code, it is checked that the caller of a method is the SNS governance canister.
* In order for this to work, the asset canister needs to know the SNS governance canister.
* The dapp canisters, including the asset canister, are handed over to the NNS as part of the request to create an SNS. Therefore at this point in time the SNS canisters do not yet exist and the SNS governance canister ID is not known.
* Therefore the asset canister needs to be updated later to "learn" the SNS governance canister ID.

Does this makes sense? It is a bit long and maybe it can be explained better, but wanted to make sure you have an answer today.

The general overview of deploying an asset canister during an SNS launch is as follows:
- First, the asset canister must be created with or upgraded to a Wasm file from [dfx 0.14.1+](https://github.com/dfinity/sdk/blob/release-0.14.1/src/distributed/assetstorage.wasm.gz).
- Then, the dapp should hand control of the asset canister over to the SNS by setting the following permissions:

>     - The SNS governance canister should be given all permissions before handing the canister over. The SNS governance canister can be manually added to all necessary permissions, including `ManagePermissions`, or it can call the `take_ownership` method to grant the necessary permissions.

Reviewer 3: `take_ownership` only grants `Commit` permissions to the caller, not `ManagePermissions`.

>     - The SNS governance canister should be given all permissions before handing the canister over. The SNS governance canister can be manually added to all necessary permissions, including `ManagePermissions`, or it can call the `take_ownership` method to grant the necessary permissions.

Reviewer 3: ```suggestion
    - The SNS governance canister must be granted `Commit` and `ManagePermissions` permissions before handing the canister over, and these permissions must be revoked from any other principal.
```

      "type": "assets"
    }
```
- Then, the dapp developers hand over control of the entire dapp to the NNS, including the asset canister.
- A single NNS proposal creates the dapp's SNS canisters, initializes them, and hands the SNS's dapp canisters over to the SNS. The decentralization swap begins.

> - The SNS community then upgrades the dapp canister through an SNS proposal and sets the privileges in the dapp. 

Author: I have moved this information to the launch steps page and replaced this content with a link to that page.
